### PR TITLE
Add pipe to eio_unix

### PR DESCRIPTION
**Pull request by patricoferris**
This adds `Eio_unix.pipe` to create a connected pair of flows similar to `socketpair`, this is part of #330

### CODE REVIEW

File: lib_eio_luv/eio_luv.ml [hunk 1]
```diff
@@ -1155,6 +1156,14 @@ let rec run : type a. (_ -> a) -> a = fun main ->
             with Luv_error _ as ex ->
               discontinue k ex
           )
+          | Eio_unix.Private.Pipe sw -> Some (fun k ->
+            let r, w = Unix.pipe ~cloexec:true () in
```
Comment (haesbaert) [inline lib_eio_luv/eio_luv.ml:1160]: libuv has pipes, can we use that ?
https://github.com/aantron/luv/blob/0ed668e87ca55d65f4f84422b0dd3381701c71b0/src/pipe.mli#L40

Comment (patricoferris) [reply lib_eio_luv/eio_luv.ml:1160]: Thanks for the speedy review! :))

Yes, I think so but I had to specify no read and write flags (see [uv_pipe](http://docs.libuv.org/en/v1.x/pipe.html#c.uv_pipe)) so it ended up being

```ocaml
let r, w = Luv.Pipe.pipe ~read_flags:[] ~write_flags:[] () |> or_raise in
```

Do you think that's okay? (They're opened with cloexec `true`).

Comment (haesbaert) [reply lib_eio_luv/eio_luv.ml:1160]: I think it's fine, the only flag it accepts is `nonblock`.

File: tests/flow.md [hunk 2]
```diff
@@ -118,3 +118,25 @@ Copying from src using `Read_source_buffer`:
...
+Writing to and reading from a pipe.
+
+```ocaml
+# Eio_main.run @@ fun env ->
+  Switch.run @@ fun sw ->
+  let r, w = Eio_unix.pipe sw in
+  Eio.Fiber.both
+    (fun () ->
+      let buf = Cstruct.create 32 in
+      let r = Eio.Flow.read r buf in
```
Comment (talex5) [inline tests/flow.md:133]: In theory, this is not reliable (though it seems unlikely to fail). Should use `read_exact` here, or `Buf_read`.

Comment (patricoferris) [reply tests/flow.md]: Indeed thanks! I've rebased to tidy up the history and used `read_exact`.

**Comment by talex5** [comment]
Thanks!

**Pull request merged**